packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -324,26 +322,46 @@ if (__DEV__) {
   ReactStrictModeWarnings.flushLegacyContextWarning = () => {
     ((pendingLegacyContextWarning: any): FiberToFiberComponentsMap).forEach(
       (fiberArray: FiberArray, strictRoot) => {
-        const uniqueNames = new Set();
+        const componentStacks = new Map();
+
         fiberArray.forEach(fiber => {
-          uniqueNames.add(getComponentName(fiber.type) || 'Component');
+          const componentName = getComponentName(fiber.type) || 'Component';
+          const componentStack = getStackByFiberInDevAndProd(fiber);
+          let count = 0;
+          if (componentStacks.has(componentStack)) {
+            ({count} = (componentStacks.get(componentStack): any));
+          }
+          // Increase count by 1
+          componentStacks.set(componentStack, {
+            count: count + 1,
+            name: componentName,
+          });
           didWarnAboutLegacyContext.add(fiber.type);
         });
 
-        const sortedNames = setToSortedString(uniqueNames);
-        const strictRootComponentStack = getStackByFiberInDevAndProd(
-          strictRoot,
-        );
+        const stacks = Array.from(componentStacks.keys());
+        // Find the most frequently found component stack and use that
+        let currentCount = 0;
+        let mostFrequentStack = null;
+
+        for (let i = 0; i < stacks.length; i++) {
+          const stack = stacks[i];
+          let {count} = (componentStacks.get(stack): any);
+          if (count > currentCount || mostFrequentStack === null) {
+            currentCount = count;
+            mostFrequentStack = stack;
+          }
+        }
 
         console.error(
           'Legacy context API has been detected within a strict-mode tree.' +
             '\n\nThe old API will be supported in all 16.x releases, but applications ' +
             'using it should migrate to the new version.' +
-            '\n\nPlease update the following components: %s' +
+            '\n\nPlease update the following component: %s' +
```

Comment:
Soo.. I'm not sure about this. I think it's still valuable to see a list of all components so you have a sense of how many there are to go. But with the stack pointing to the most frequent one.

Reply:
Agreed. Only showing one at a time could be pretty frustrating- like whack-a-mole, where you begin to wonder if the list will ever end. I think showing *all* component names (like before) is probably best.

Not sure if "the most frequent one" is necessarily valuable since you'll need to fix them all if you want to be CM compatible, but showing the location of [whichever] one would probably be a nice addition.

Reply:
I agree. I’ll add it back, with the most frequent component appearing first.

Reply:
This is resolved. I'd appreciate another look :)